lib/phoenix_html/form.ex

```diff
@@ -738,6 +738,11 @@ defmodule Phoenix.HTML.Form do
         opts
       end
 
+    opts =
+      Keyword.update(opts, :checked, false, fn c ->
+        boolean_attribute("checked", c)
```

Comment:
Question: Why do we need this? Can't we just pass the boolean forward?

Reply:
Passing the boolean forward will mean the naked attribute version will be output instead. I added the boolean_attribute() helper to retain the current behavior for `form_for`, `select`, `radio_button`, etc. They currently emit the boolean form `checked="checked"` and I thought it would be less impactful for that not to change. That is still a valid boolean form. If we just pass the boolean forward, then many of the current tests need to be updated, and people might be surprised that the html generation changed. Either works -- this is just less disruptive, I feel.

Reply:
I see. Let's go ahead and rip the band-aid, otherwise people may be confused why some attributes work like that and others do not. :)